site/pages/api/publish-edit.api.ts

```diff
@@ -145,6 +150,28 @@ const publishEdit = async (req: NextApiRequest, res: NextApiResponse) => {
         }
 
         cleardownCookies(req, res);
+
+        if (session.isOrgStaff && !template) {
+            const orgAdminsForAllOrgs = await getAllUsersInGroup("org-admins");
+
+            const orgAdminEmailsForStaffOrg = getOrgAdminEmailsForStaffUserOrg(orgAdminsForAllOrgs, session.orgId);
+
+            if (!orgAdminEmailsForStaffOrg || orgAdminEmailsForStaffOrg.length === 0) {
+                logger.error(`No organisation admins found for Org Id ${session.orgId}`);
+                redirectTo(res, DASHBOARD_PAGE_PATH);
+                return;
+            }
+
+            const disruptionApprovalEmail = createDisruptionApprovalEmail(
+                validatedDisruptionBody.data.summary,
+                validatedDisruptionBody.data.description,
+                session.name,
+                validatedDisruptionBody.data.disruptionId,
+                orgAdminEmailsForStaffOrg,
+            );
+            await sesClient.send(disruptionApprovalEmail);
```

Comment:
Conscious that with increasing steps in publish flow - the response returned to the user could be delayed. Can we refactor the above block of code and invoke it asynchronously?

Reply:
refactored to create a separate function sendDisruptionApprovalEmail() that is invoked asynchronously